src/main/java/duke/task/Storage.java

```diff
@@ -0,0 +1,122 @@
+package duke.task;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+import duke.exception.InvalidParamException;
+import duke.main.TaskList;
+
+/**
+ * Encapsulates a storage that has methods to save Tasks from a TaskList in the hard disk,
+ * as well as loading that saved data back into the TaskList upon running Duke.
+ */
+public class Storage {
+
+    /** The path of the text file used to store the Tasks from a TaskList */
+    private final String FILEPATH = "tasklist.txt";
+    /** The file used to store the Tasks from a TaskList */
+    private final File SAVE_FILE = new File(FILEPATH);
+    /** A distinct string separating pieces of information saved in the save text file */
+    private final String SEPARATOR = "~SEPARATION_STRING~";
+
+    /** TaskList that this Storage will copy to and from */
+    private TaskList taskList;
+
+    /**
+     * Constructs a Storage instance that acts on the given TaskList.
+     *
+     * @param taskList TaskList that this Storage will copy to and from.
+     */
+    public Storage(TaskList taskList) {
+        try {
+            this.taskList = taskList;
+            SAVE_FILE.createNewFile();
+        } catch (java.io.IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Copies Tasks from the TaskList in order and saves them in a save text file.
+     * Each Task is saved on 1 line in the order and format:
+     * Status(1 for done, 0 for not done), Initial of Task, Description, Extra information
+     * The above pieces of information are separated by the SEPARATOR string.
+     */
+    public void writeToFile() {
+        try {
+            StringBuilder toBeWritten = new StringBuilder();
+            FileWriter fileWriter = new FileWriter(FILEPATH);
+
+            for (Task task : taskList) {
+                char initialOfTask = task.toString().charAt(1);
+                String description = task.getDescription();
+
+                // Add a marker to tell if its done or not
+                toBeWritten.append(task.getIsDone() ? "1" : "0").append(SEPARATOR);
+
+                //Add the task type and description
+                toBeWritten.append(initialOfTask).append(SEPARATOR).append(description);
+
+                switch (initialOfTask) {
+                case 'T':
```

Comment:
```
private static final char TASK_TYPE = 'T';
```
Should this magic character be moved to top by using named constant or use enum? 🤔
And I noticed the same issue in several other places below.

Reply:
Hey I tried taking this into consideration and changing it to an enum but apparently it doesn't work. Switch block cases require 'constant expressions', and even when I set the enum field of `initial` to be `final`, it did not work. So I guess I'll just leave it as it is. Thanks for the suggestion!